packer/sagebio/sagebio-base-ubuntu-bionic-LATEST/playbook.yaml

```diff
@@ -17,13 +17,18 @@
           - "curl"
         state: present
 
-    # awscli is on aws linux2 distro by default
-    - name: install awscli
+    - name: Prep snap classic
+      shell: "[ -x /snap ] || yum install -y epel-release snapd && systemctl enable --now snapd.socket && ln -s /var/lib/snapd/snap /snap || echo ''" #install via yum on CentOS if DNE
+      register: snap_installed
+
+    # awscli and amazon-ssm-agent are on aws linux2 by default
+    - name: Install awscli and amazon-ssm-agent if available in Snappy
       snap:
         name:
           - aws-cli
+          - amazon-ssm-agent
         classic: yes
-      when: ansible_distribution == "Ubuntu"
+      when: "not snap_installed.stdout"
```

Comment:
would be nicer to use package_facts to check that snap is installed.

Reply:
I think we need to check that a symlink at /snap exists, rather than check package facts. This symlink 'enables classic snaps': https://snapcraft.io/docs/installing-snap-on-centos